### PR TITLE
dev/core#781 Additional email greetings in message templates

### PR DESCRIPTION
**Pull request by magnolia61**
Overview

For many standard workflow message templates the greeting was already changed to {contact.email_greeting}. This PR fixes the remaining templates.

**Before**

Not all standard workflow message templates were using {contact.email_greeting}

**After**

All standard workflow message templates are using {contact.email_greeting}

**Technical Details**

Additionally in the participant_confirm template a missing {ts}{/ts} was fixed for the confirmation Url.

This PR also serves as the upgrader script for the following other PR's in 5.20alpha1:
https://github.com/civicrm/civicrm-core/pull/15503 ( Update 'Thank You' & other corrections) (https://lab.civicrm.org/dev/core/issues/1316)
https://github.com/civicrm/civicrm-core/pull/15467 (Remove template refs to 'print for your records' (https://lab.civicrm.org/dev/core/issues/371)
https://github.com/civicrm/civicrm-core/pull/15513 (include displayname in subject) https://lab.civicrm.org/dev/core/issues/1320

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
@magnolia61 in the case of the 2 test failures the tests need to be updated to check for the right string now that that has changed

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by eileenmcnaughton** [comment]
@magnolia61 ug - I  merged in a related PR & now this won't apply - can you ``` git pull --rebase ``` & ``` git push -f ```?

**Comment by eileenmcnaughton** [comment]
looks like a smarty error has snuck in 

api_v3_PaymentTest::testRefundEmailReceipt with data set #0 ('.')
Failure in api call for Payment sendconfirmation:  Smarty error: [in string:{assign var=&quot;greeting&quot; value=&quot;Dear Anthony&quot;}{if $greeting}{$greeting},{/if}

**Comment by eileenmcnaughton** [comment]
@magnolia61 still failing :-(

**Comment by magnolia61** [comment]
> @magnolia61 still failing :-(

think I've fond the culprit. An extra {/if}. I expect it to pass now.

**Comment by eileenmcnaughton** [comment]
@magnolia61 still some update fails & an eroneous .swp file in there. 

It might be easier to drop out the upgrade part & do just the tpl files in this commit & get them merged & follow up with the update - just thinking it's been a pain to get the tpl files to this point & it would be nice to get that part merged & focus on the other part

**Comment by magnolia61** [comment]
> 
> 
> @magnolia61 still some update fails & an eroneous .swp file in there.
> 
> It might be easier to drop out the upgrade part & do just the tpl files in this commit & get them merged & follow up with the update - just thinking it's been a pain to get the tpl files to this point & it would be nice to get that part merged & focus on the other part

'All checks have passed'

This pr will make it possible also for these PR's to be included in this rounds template upgrade
https://github.com/civicrm/civicrm-core/pull/15467
https://github.com/civicrm/civicrm-core/pull/15503
https://github.com/civicrm/civicrm-core/pull/15513

**Comment by eileenmcnaughton** [comment]
@magnolia61 this file needs to be dropped from the commit 
https://github.com/civicrm/civicrm-core/pull/15491/files#diff-ced36ec4b317ada544a992026a0698d1 (not sure if it;'s the only .swp file that snuck in)

**Comment by magnolia61** [comment]
> @magnolia61 this file needs to be dropped from the commit
> https://github.com/civicrm/civicrm-core/pull/15491/files#diff-ced36ec4b317ada544a992026a0698d1 (not sure if it;'s the only .swp file that snuck in)

removed the naughty .swp file ;-)

File: xml/templates/message_templates/payment_or_refund_notification_text.tpl [hunk 1]
```diff
@@ -1,5 +1,6 @@
-{if $emailGreeting}{$emailGreeting},
-{/if}{if $isRefund}
+{assign var="greeting" value="{contact.email_greeting}"}{if $greeting}{$greeting},{/if}
```
Comment (eileenmcnaughton) [inline xml/templates/message_templates/payment_or_refund_notification_text.tpl]: I don't think we should change this one - it already has emailGreeting assigned so we should use it rather than smarty magic

File: xml/templates/message_templates/payment_or_refund_notification_html.tpl [hunk 2]
```diff
@@ -20,9 +20,9 @@
   <!-- END HEADER -->
 
   <!-- BEGIN CONTENT -->
-   {if $emailGreeting}<tr><td>{$emailGreeting},</td></tr>{/if}
```
Comment (eileenmcnaughton) [inline xml/templates/message_templates/payment_or_refund_notification_html.tpl]: per below we can keep this one

File: tests/phpunit/api/v3/PaymentTest.php [hunk 3]
```diff
@@ -181,7 +181,7 @@ public function testPaymentEmailReceipt() {
     $this->callAPISuccess('Payment', 'sendconfirmation', ['id' => $payment['id']]);
     $mut->assertSubjects(['Payment Receipt - Annual CiviCRM meet']);
     $mut->checkMailLog([
-      'Dear Anthony,',
+      'Dear Anthony',
```
Comment (eileenmcnaughton) [inline tests/phpunit/api/v3/PaymentTest.php]: hmm - the comma shouldn't be disappearing - still I think this is the same template I said we shouldn't change

**Comment by eileenmcnaughton** [comment]
this is very close now - just a couple of minor comments

File: tests/phpunit/api/v3/PaymentTest.php (hunk 3 above)
Comment (magnolia61) [reply tests/phpunit/api/v3/PaymentTest.php]: I though it would serve consistency as most tests do not have the comma. Would you suppose the other way around? Add the comma to the other tests?

File: xml/templates/message_templates/payment_or_refund_notification_text.tpl (hunk 1 above)
Comment (magnolia61) [reply xml/templates/message_templates/payment_or_refund_notification_text.tpl]: I thought this would help consistency. This is the only template {$emailGreeting} is used. Not sure why that is. I think code is easier to read if it is consistent. If you still think it;s better to remove, I will do so.

Comment (eileenmcnaughton) [reply xml/templates/message_templates/payment_or_refund_notification_text.tpl]: @magnolia61 yes, because  you  would also need to remove where emailGreeting is assigned to the template to remove it from the template

**Comment by magnolia61** [comment]
test this please

**Comment by eileenmcnaughton** [comment]
@magnolia61 arg - it needs a rebase :-(

File: CRM/Event/BAO/Event.php [hunk 4]
```diff
@@ -1172,6 +1172,7 @@ public static function sendMail($contactID, $values, $participantId, $isTest = F
           'customPost' => $profilePost[0],
           'customPost_grouptitle' => $customPostTitles,
           'participantID' => $participantId,
+          'contactID' => $contactID,
```
Comment (eileenmcnaughton) [inline CRM/Event/BAO/Event.php]: Ug you have pulled in some unrelated stuff

Comment (magnolia61) [reply CRM/Event/BAO/Event.php]: test all passed now. Are you able to merge this?

**Comment by eileenmcnaughton** [comment]
Yes - I think we are there - good to see so many test changes needed - shows test coverage
